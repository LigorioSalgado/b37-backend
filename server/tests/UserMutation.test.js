jest.mock('../services/UserService.js', () => jest.requireActual('../services/__mocks__/UserService.js'));
const { createTestClient } = require('apollo-server-testing');
const gql = require('graphql-tag');
const { ApolloServer } = require('apollo-server');
const  { schema } = require('../index');
//const { importSchema } = require('graphql-import');
//7const resolvers =  require('../resolvers');
//const typeDefs = importSchema('../schema.graphql');

const CREATE_USER = gql`
mutation createUser($Userdata:UserAdd!){
  newUser(data:$Userdata){
    _id
    first_name
    last_name
  }
}

`;

const GET_USERS = gql`
query getUsers{
  allUsers{
    _id
    email
    first_name
    last_name
    
  }
}

`;

test('Create User', async() => {
	const server = new ApolloServer({
		schema
	});
	const { mutate } = createTestClient(server);

	const res =  await mutate({
		mutation: CREATE_USER,
		variables: { Userdata:{
			email:'prueba@prueba.com',
			first_name:'prueba',
			last_name: 'prueba',
			password:'prueba'
		}}
	});

	expect(res).toMatchSnapshot();
	expect(res.data.newUser).toHaveProperty('_id');
});


test('Not Create User', async() => {
	const server = new ApolloServer({
		schema
	});
	const { mutate } = createTestClient(server);

	const res =  await mutate({
		mutation: CREATE_USER,
		variables: { Userdata:{
			last_name: 'prueba',
			password:'prueba'
		}}
	});

	expect(res).toHaveProperty('errors');
});

test('Create User', async() => {
	const server = new ApolloServer({
		schema
	});
	const { mutate } = createTestClient(server);

	const res =  await mutate({
		mutation: CREATE_USER,
		variables: { Userdata:{
			email:'prueba@prueba.com',
			first_name:'prueba',
			last_name: 'prueba',
			password:'prueba'
		}}
	});

	expect(res).toMatchSnapshot();
	expect(res.data.newUser).toHaveProperty('_id');
});

test('Get Users', async() => {
	const server = new ApolloServer({
		schema
	});
	const { query } = createTestClient(server);

	const res =  await query({
		mutation: GET_USERS,});

	expect(res).toMatchSnapshot();
});

