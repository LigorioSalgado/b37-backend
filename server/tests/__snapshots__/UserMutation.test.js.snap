// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create User 1`] = `
Object {
  "data": Object {
    "newUser": Object {
      "_id": "5e2bb4004809b000d6141c73",
      "first_name": "Edwin",
      "last_name": "Salgado",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Create User 2`] = `
Object {
  "data": Object {
    "newUser": Object {
      "_id": "5e2bb4004809b000d6141c73",
      "first_name": "Edwin",
      "last_name": "Salgado",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Get Users 1`] = `
Object {
  "data": Object {
    "allUsers": Array [
      null,
    ],
  },
  "errors": Array [
    [GraphQLError: Cannot return null for non-nullable field User._id.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
